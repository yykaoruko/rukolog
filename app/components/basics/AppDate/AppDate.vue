<template>
  <time :datetime="datetimeValue" class="app-date">
    <span class="app-date__date">
      {{ date.year }}/{{ date.month }}/{{ date.date }}
    </span>
  </time>
</template>

<script lang="ts">
import Vue from 'vue';
import {
  splitDatetime,
  formatToDatetimeAttribute,
} from '@/helpers/date/date.helper';
import { isLogDate } from '@/helpers/validator/validator.helper';
export default Vue.extend({
  name: 'AppDate',
  props: {
    time: {
      type: Boolean,
      default: false,
    },
    datetime: {
      type: String,
      required: true,
      validator(value) {
        return isLogDate(value);
      },
    },
  },
  computed: {
    date() {
      return splitDatetime(this.datetime);
    },
    datetimeValue() {
      return formatToDatetimeAttribute(this.datetime);
    },
  },
});
</script>

<style lang="scss" scoped>
.app-date {
  color: inherit;
}
</style>
