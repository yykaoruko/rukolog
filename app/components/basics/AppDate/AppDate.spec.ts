import Vue from 'vue';
import { shallowMount, Wrapper } from '@vue/test-utils';
import AppDate from './AppDate.vue';

describe('AppDate', () => {
  describe('mount', () => {
    let wrapper: Wrapper<Vue>;

    beforeEach(() => {
      wrapper = shallowMount(AppDate, {
        propsData: {
          datetime: '2019-10-07',
        },
      });
    });

    it('mountable', () => {
      expect(wrapper.isVueInstance()).toBe(true);
    });

    it('snapshot', () => {
      expect(wrapper.element).toMatchSnapshot();
    });
  });
});
