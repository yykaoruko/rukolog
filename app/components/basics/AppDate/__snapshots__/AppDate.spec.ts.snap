// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppDate mount snapshot 1`] = `
<time
  class="app-date"
  datetime="2019-10-07T12:00:00"
>
  <span
    class="app-date__date"
  >
    
    19/10/07
  
  </span>
</time>
`;
